### PR TITLE
Weather App

### PR DESCRIPTION
**Pull request by Shaher-11**
This the weather application  build with javascript , an API from OpenWeather as well taking advantage of webpack in the project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,54 +1,36 @@
...
 
-### Prerequisites
+- HTML
+- Bootstrap
+- JavaScript
+- Open Weather API
+- Lots of love :heart:
 
-To get this project up and running locally, you must already have ruby and the necessary gems installed on your computer
+## Live Demo
```
Comment (addod19) [inline README.md:19]: Kindly add instructions about how to clone and run the project locally in the README file :+1:

File: src/ui.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/* eslint-disable import/no-cycle */
```
Comment (addod19) [inline src/ui.js]: Kindly avoid the inline ESLint disabling rule, this is equally the same as altering or changing our linter files. Let us fix it

File: src/ui.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { fetchWeatherData } from './weather';
+/* eslint-enable import/no-cycle */
```
Comment (addod19) [inline src/ui.js]: Kindly avoid the inline ESLint disabling rule, this is equally the same as altering or changing our linter files. Let us fix it

File: src/weather.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable import/no-cycle, import/no-cycle, import/prefer-default-export */
```
Comment (addod19) [inline src/weather.js]: Kindly remote comment and avoid the inline ESLint disabling rule, this is equally the same as altering or changing our linter files. Let us fix it

File: src/weather.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    insertData(data);
+    changeAppearance(weatherData.main.temp, units);
+  }
+  getData();
+};
+
+fetchWeatherData();
+
+export { fetchWeatherData };
+/* eslint-enable import/no-cycle, import/no-cycle, import/prefer-default-export */
```
Comment (addod19) [inline src/weather.js]: Kindly avoid the inline ESLint disabling rule, this is equally the same as altering or changing our linter files. Let us fix it

File: package.json [hunk 6]
```diff
@@ -2,9 +2,11 @@
   "name": "waetherjs",
   "version": "1.0.0",
   "description": "![](https://img.shields.io/badge/Microverse-blueviolet)",
-  "main": "main.js",
+  "main": "index.js",
+  "private": "true",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (addod19) [inline package.json]: Kindly remove the "test" script and replace it with the "dev" script, so we can run both the development and production scripts :+1:

File: src/weather.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable import/no-cycle, import/no-cycle, import/prefer-default-export */
+import { insertData, changeAppearance } from './ui';
+
+const fetchWeatherData = (location = 'Cairo', units = 'metric') => {
+  const url = `http://api.openweathermap.org/data/2.5/weather?q=${location}&units=${units}&APPID=c0200200bfb501d04806a813e7fcf708`;
+
+  async function getData() {
```
Comment (addod19) [inline src/weather.js]: Kindly use the ES6 arrow syntax instead :smile:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi, @Shaher-11 great job on this project :+1: Nice work separating the application logic from the UI components :+1: This makes testing easier :fireworks: Great job implementing the error handling for async/await :fireworks: However, there are a few comments we need to address before proceeding. Kindly check the comments below:

- Kindly use the right linter files recommended by the students of Microverse. You can find the link here [.eslintrc.json](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json)
- There is a missing functionality; there should be a background change when data is received
- Kindly remove the `storage.js` file since we are not using it in our project

Once these are addressed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/weather.js (hunk 7 above)
Comment (Shaher-11) [reply src/weather.js]: that's An async function declaration

Comment (rukundoeric) [inline src/weather.js]: Fot this async function, use this format 
```Javascript 
      const getData = async () => {
          // Code here
      }
```

File: README.md [hunk 8]
```diff
@@ -1,54 +1,53 @@
...
+
+- HTML
+- Bootstrap
+- JavaScript
+- Open Weather API
 - Lots of love :heart:
 
+## Live Demo
+
+[View Live](https://fabuloustours.netlify.app/)
```
Comment (rukundoeric) [inline README.md]: This link is not for this project, kindly change it.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 


- ### To Improve
  - [ ] [MANDATORY] The live-demo link you put in your README is for a different project.
  - [ ] [MANDATORY] As it is mentioned by the previous review, eslint errors are not to be ignored, instead try to fix them.
      - `import/no-cycle`: To fix this error, make sure you don't have two modules importing resources from each other. in your case, `ui.js` is importing `fetchWeatherData` from `weather.js` and `weather.js` is importing  `insertData and changeAppearance` from `ui.js`. kindly, try to fix this.
      - `import/prefer-default-export`: In case you have only one function in the module, it is better to export the function as default.
  - [ ] [MANDATORY] I tried to run the app, but there were some errors trying to access the API. Check out this [Video](https://www.loom.com/share/4bf8efafce7d4afe92072f248170b8a6) to see what is happening.  Kindly fix that and don't forget to make sure that the app changes background according to the received data.    

_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by uche-inyama — APPROVED** [review]
## Approved

### Hi Shaher Mahmoud :slightly_smiling_face: 

We are good here, all the requested changes were implemented. There are no disabled eslint comments, the application is running correctly locally and the link to the live demo is working.

### Go on with the next task.

###  Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Pull request merged**